pallets/pool-registry/src/benchmarking.rs

```diff
@@ -11,37 +11,74 @@
+	Pool, PoolDetailsOf, TrancheOf,
+};
+use sp_runtime::{
+	traits::{One, Zero},
+	Perquintill,
+};
 use sp_std::vec;
 
 use super::*;
 
+const CURRENCY: u128 = 1_000_000_000_000_000;
+const MAX_RESERVE: u128 = 10_000 * CURRENCY;
+const MINT_AMOUNT: u128 = 1_000_000 * CURRENCY;
+
+const SECS_PER_HOUR: u64 = 60 * 60;
+const SECS_PER_DAY: u64 = 24 * SECS_PER_HOUR;
+const SECS_PER_YEAR: u64 = 365 * SECS_PER_DAY;
+
+const TRANCHE: TrancheIndex = 0;
+
 const POOL: u64 = 0;
 
 benchmarks! {
 	where_clause {
 	where
-		T: Config<PoolId = u64>,
+		T: Config<PoolId = u64,
+			  TrancheId = [u8; 16],
+			  Balance = u128,
+			  CurrencyId = CurrencyId>,
+		T: pallet_pool_system::Config<PoolId = u64,
+			  TrancheId = [u8; 16],
+			  Balance = u128,
+			  CurrencyId = CurrencyId,
+			  EpochId = PoolEpochId>,
+		<<T as frame_system::Config>::Lookup as sp_runtime::traits::StaticLookup>::Source:
+			From<<T as frame_system::Config>::AccountId>,
+		<T as pallet_pool_system::Config>::Permission: Permissions<T::AccountId, Ok = ()>,
+		<T as Config>::ModifyPool: PoolMutate<<T as frame_system::Config>::AccountId, <T as pallet::Config>::PoolId, TrancheInput = TrancheInput<<T as pallet_pool_system::Config>::Rate, <T as pallet_pool_system::Config>::MaxTokenNameLength, <T as pallet_pool_system::Config>::MaxTokenSymbolLength>>,
+	}
+	register {
+		let n in 1..<T as pallet_pool_system::Config>::MaxTranches::get();
+		let caller: <T as frame_system::Config>::AccountId = create_admin::<T>(0);
+		let tranches = build_bench_input_tranches::<T>(n);
+		let origin = RawOrigin::Signed(caller.clone());
+		prepare_asset_registry::<T>();
+	}: register(origin, caller, POOL, tranches.clone(), CurrencyId::AUSD, MAX_RESERVE, None)
+	verify {
+		let pool = get_pool::<T>();
+		assert_input_tranches_match::<T>(pool.tranches.residual_top_slice(), &tranches);
+		assert_eq!(pool.reserve.available, Zero::zero());
+		assert_eq!(pool.reserve.total, Zero::zero());
+		assert_eq!(pool.parameters.min_epoch_time, T::DefaultMinEpochTime::get());
+		assert_eq!(pool.parameters.max_nav_age, T::DefaultMaxNAVAge::get());
+		assert_eq!(pool.metadata, None);
 	}
-
-	// create {
-	// 	let n in 1..T::MaxTranches::get();
-	// 	let caller: T::AccountId = create_admin::<T>(0);
-	// 	let tranches = build_bench_input_tranches::<T>(n);
-	// 	let origin = RawOrigin::Signed(caller.clone());
-	// 	prepare_asset_registry::<T>();
-	// }: create(origin, caller, POOL, tranches.clone(), CurrencyId::AUSD, MAX_RESERVE, None)
-	// verify {
-	// 	let pool = get_pool::<T>();
-	// 	assert_input_tranches_match::<T>(pool.tranches.residual_top_slice(), &tranches);
-	// 	assert_eq!(pool.reserve.available, Zero::zero());
-	// 	assert_eq!(pool.reserve.total, Zero::zero());
-	// 	assert_eq!(pool.parameters.min_epoch_time, T::DefaultMinEpochTime::get());
-	// 	assert_eq!(pool.parameters.max_nav_age, T::DefaultMaxNAVAge::get());
-	// 	assert_eq!(pool.metadata, None);
-	// }
```

Comment:
We still need to write a seperate case for 
* `update`
   * I think here it is enough to simply use the worst case of update and execute
* `execute_update`